pkg/minishift/openshift/version/openshift_versions.go

```diff
@@ -17,48 +17,136 @@ limitations under the License.
 package version
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
-
-	"github.com/google/go-github/github"
-	"github.com/minishift/minishift/pkg/minishift/util"
-	githubutil "github.com/minishift/minishift/pkg/util/github"
-	"github.com/pkg/errors"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
 )
 
-const githubOwner = "openshift"
-const githubRepo = "origin"
+type ImageTags struct {
+	Count    int         `json:"count"`
+	Next     string      `json:"next"`
+	Previous interface{} `json:"previous"`
+	Results  []Results   `json:"results"`
+}
+
+type Results struct {
+	Name        string      `json:"name"`
+	FullSize    int         `json:"full_size"`
+	Images      []ImageInfo `json:"images"`
+	ID          int         `json:"id"`
+	Repository  int         `json:"repository"`
+	Creator     int         `json:"creator"`
+	LastUpdater int         `json:"last_updater"`
+	LastUpdated time.Time   `json:"last_updated"`
+	ImageID     interface{} `json:"image_id"`
+	V2          bool        `json:"v2"`
+}
 
-func PrintOpenShiftVersionsFromGitHub(output io.Writer) {
-	PrintOpenShiftVersions(output)
+type ImageInfo struct {
+	Size         int         `json:"size"`
+	Architecture string      `json:"architecture"`
+	Variant      interface{} `json:"variant"`
+	Features     interface{} `json:"features"`
+	Os           interface{} `json:"os"`
+	OsVersion    interface{} `json:"os_version"`
+	OsFeatures   interface{} `json:"os_features"`
 }
 
-func PrintOpenShiftVersions(output io.Writer) {
-	versions, err := getVersions()
+func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
+	resp, err := getResponseBody("https://registry.access.redhat.com/v1/repositories/openshift3/ose/tags")
 	if err != nil {
-		fmt.Println(err.Error())
+		fmt.Println("Error Occured", err)
 		return
 	}
-	fmt.Fprint(output, "The following OpenShift versions are available: \n")
-
-	for _, version := range versions {
-		if util.ValidateOpenshiftMinVersion(*version.TagName) {
-			fmt.Fprintf(output, "\t- %s\n", *version.TagName)
+	defer resp.Body.Close()
```

Comment:
I think the problem is that this function does too much. It is almost identical to the upstream version, apart from parsing the versions. 

So why don't you extract a function which returns a slice of version strings and pass this slice to a method which can be shared between the upstream and downstream print method.